core/src/main/java/com/gentics/mesh/core/data/container/impl/NodeGraphFieldContainerImpl.java

```diff
@@ -390,8 +390,8 @@ private boolean updateWebrootPathInfo(Node node, GraphFieldContainerEdge edge, S
 		String segment = node.getPathSegment(branchUuid, type, getLanguageTag());
 
 		// The webroot uniqueness will be checked by validating that the string [segmentValue-branchUuid-parentNodeUuid] is only listed once within the given
-		// specific index for (drafts or published nodes)
-		if (segment != null) {
+		// specific index for (drafts or published nodes)Segment field should also be set, otherwise the segment info is treated as outdated and is subject to reset.
+		if (segment != null && StringUtils.isNotBlank(segmentFieldName)) {
```

Comment:
Could we make the test for the segmentFieldName earlier? Like it is done in the branch hotfix-1.9.x-webroot-bypass

Reply:
Then we are completely skipping the reset of `setSegmentInfo()`, which is not good for the case of segment field name removal. Or do you mean placing the `setSegmentInfo()` to the early return as well?

Reply:
Ah, I see. We still need to go to the "else" branch, if there is no segment field name. That's ok then.